README.md

```diff
@@ -2,63 +2,211 @@
+
+The extension does require the following to be installed before use:
+
+- VS Code v1.61 or greater
+- Terraform v0.12 or greater
+
+## Platform Support
+
+The extension should work anywhere VS Code itself and Terraform 0.12 or higher is supported. Our test matrix includes the following:
+
+- Windows Server 2019 with Terraform v1.1
+- macOS 10.15 with Terraform v1.1
+- Ubuntu 20.04 with Terraform v1.1
+
+Intellisense, error checking and other language features are supported for Terraform v0.12 and greater.
+
+Syntax highlighting supports Terraform v1.0 and greater. Syntax support for v0.12 and v0.15 is on a best effort basis.
+
+## Usage
+
+### VS Code Workspace support
+
+It is a common pattern to have seperate folders containing related Terraform configuration that are not contained under one root folder. For example, you have a main Terraform folder containing the configuration for a single application and several module folders containing encapsualted code for configuring different parts of component pieces. You could open each folder in a separate VS Code window, and bounce between each window to author your changes.
+
+A better approach is to use [VS Code Workspaces](https://code.visualstudio.com/docs/editor/workspaces). Using our exmaple above, open the main Terraform folder first, then use Add Folder to workspace to add the dependent module folders. A single VS Code window is used and all Terrraform files are available to author your changes. This uses a single terraform-ls process that has an understanding of your entire project, allowing you to use features like `Go to Symbol` and `Reference counts` across your project.
+
+### Single file support
+
+Opening a single Terraform configuration file inside VS Code is currently not supported. We see this approach most commonly attempted by users of terminal editors like vim, where it is common to edit a single file at a time.
+
+The Terraform VS Code extension works best when it has the full context of a Terraform project where it can parse the referenced files and provide the expected advanced language features.
+
+The recommend workflow is to instead  open the containing folder for the desired Terraform file inside a single VS Code editor window, then navigate to the desired file. This seems counter-intuitive when you only want to edit a single file, but this allows the extension to understand the Terraform setup you are using and provide accurate and helpful intellisense, error checking, and other language features.
+
+### Refresh Intellisense
+
+To provide the extension with an up-to-date schema for the Terraform providers used in your configuration:
+
+1. Open a terminal and change the current working directory to your desired workspace and/or the root folder containing your Terraform files.
+1. Perform a `terraform init` from the terminal.
+1. Open VS Code in the current working directory by typing `code .`
```

Comment:
FWIW the language server should be able to pick up new providers automatically whenever anything changes in `.terraform`, so `init` should actually be sufficient here - there should be no need to restart VSCode.

Reply:
Updated